jfrog-cli/artifactory/commands/curl/curl.go

```diff
@@ -0,0 +1,247 @@
+package curl
+
+import (
+	"errors"
+	"fmt"
+	gofrogcmd "github.com/jfrog/gofrog/io"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/utils/config"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"io"
+	"os/exec"
+	"strings"
+)
+
+func Execute(args []string) error {
+	// Get curl execution path.
+	execPath, err := exec.LookPath("curl")
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	// Create curl command.
+	command := &CurlCommand{ExecutablePath: execPath, Arguments: args}
+
+	// Get server-id, remove from the command if exists.
+	serverIdValue, err := command.getAndRemoveServerIdFromCommand()
+	if err != nil {
+		return err
+	}
+
+	// Get Artifactory details for this ID.
+	artDetails, err := config.GetArtifactorySpecificConfig(serverIdValue)
+	if err != nil {
+		return err
+	}
+
+	// If the command already include credentials flag, return an error.
+	if command.isCredentialsFlagExists() {
+		return errorutils.CheckError(errors.New("Curl command must not include credentials flag (-u or --user)."))
+	}
+
+	// Get target url for the curl command.
+	urlIndex, targetUrl, err := command.buildCommandUrl(artDetails.Url)
+	if err != nil {
+		return err
+	}
+
+	// Replace url argument with complete url.
+	command.Arguments[urlIndex] = targetUrl
+
+	log.Debug(fmt.Sprintf("Executing curl command: '%s -u***:***'", strings.Join(command.Arguments, " ")))
+	// Add credentials flag to Command. In case of flag duplication, the latter is used by Curl.
+	credFlag := fmt.Sprintf("-u%s:%s", artDetails.User, artDetails.Password)
+	command.Arguments = append(command.Arguments, credFlag)
+
+	// Run curl.
+	return gofrogcmd.RunCmd(command)
+}
+
+func (curlCmd *CurlCommand) buildCommandUrl(artifactoryUrl string) (urlIndex int, urlValue string, err error) {
```

Comment:
This comment is relevant for multiple places in this file.
I think that when using the ```jfrog rt curl``` command, the user is asked to specify a URI (for exampe /api/build) and not a URL (http://aaa.com/api/build). I therefore suggest changing all the variables with include *url* as part fo their name to *uri*. For example, ```urlIndex``` and ```urlValue```.

Reply:
Changed.